src/python/pants/backend/python/util_rules/lockfile_metadata.py

```diff
@@ -166,14 +162,15 @@ def _from_json_dict(
 
         return PythonLockfileMetadataV2(interpreter_constraints, requirements)
 
-    def _header_dict(self) -> dict[Any, Any]:
+    @classmethod
+    def additional_header_attrs(cls, instance: LockfileMetadata) -> dict[Any, Any]:
+        instance = cast(PythonLockfileMetadataV2, instance)
+        # Requirements need to be stringified then sorted so that tests are deterministic. Sorting
+        # followed by stringifying does not produce a meaningful result.
         return {
-            **super()._header_dict(),
-            # Requirements need to be stringified then sorted so that tests are deterministic.
-            # Sorting followed by stringifying does not produce a meaningful result.
-            "generated_with_requirements": (
-                sorted(str(i) for i in self.requirements) if self.requirements is not None else None
-            ),
+            "generated_with_requirements": sorted(str(i) for i in instance.requirements)
+            if instance.requirements is not None
+            else None,
```

Comment:
Not a big deal, but using `()` to wrap the expression makes this much more readable. Compare before and after

Reply:
Agree. Fixing, and grumbling at black for making the formatting worse :)